sqlserver/datadog_checks/sqlserver/queries.py

```diff
@@ -21,3 +21,129 @@
+        SELECT
+            AG.group_id AS availability_group,
+            AG.name AS availability_group_name,
+            AR.replica_server_name,
+            AR.failover_mode_desc,
+            AR.availability_mode_desc,
+            ADC.database_name,
+            DRS.replica_id,
+            DRS.database_id,
+            DRS.is_primary_replica,
+            DRS.database_state,
+            DRS.synchronization_state_desc,
+            DRS.log_send_queue_size,
+            DRS.log_send_rate,
+            DRS.redo_queue_size,
+            DRS.redo_rate,
+            DRS.low_water_mark_for_ghosts,
+            DRS.filestream_send_rate,
+            DRS.secondary_lag_seconds,
+            FC.cluster_name
+        FROM
+            sys.availability_groups AS AG
+            INNER JOIN sys.availability_replicas AS AR ON AG.group_id = AR.group_id
+            INNER JOIN sys.availability_databases_cluster AS ADC ON AG.group_id = ADC.group_id
+            INNER JOIN sys.dm_hadr_database_replica_states AS DRS ON AG.group_id = DRS.group_id
+                AND ADC.group_database_id = DRS.group_database_id
+                AND AR.replica_id = DRS.replica_id
+            CROSS APPLY (SELECT cluster_name FROM sys.dm_hadr_cluster) AS FC
+    """.strip(),
+    'columns': [
+        # AG - sys.availability_groups
+        {'name': 'availability_group', 'type': 'tag'},
+        {'name': 'availability_group_name', 'type': 'tag'},
+        # AR - sys.availability_replicas
+        {'name': 'replica_server_name', 'type': 'tag'},
+        {'name': 'failover_mode_desc', 'type': 'tag'},
+        {'name': 'availability_mode_desc', 'type': 'tag'},
+        # ADC - sys.availability_databases_cluster
+        {'name': 'database_name', 'type': 'tag'},
+        # DRS - sys.dm_hadr_database_replica_states
+        {'name': 'replica_id', 'type': 'tag'},
+        {'name': 'database_id', 'type': 'tag'},
+        {'name': 'ao.is_primary_replica', 'type': 'gauge'},
+        {'name': 'database_state', 'type': 'tag'},
+        {'name': 'synchronization_state_desc', 'type': 'tag'},
+        {'name': 'ao.log_send_queue_size', 'type': 'gauge'},
+        {'name': 'ao.log_send_rate', 'type': 'gauge'},
+        {'name': 'ao.redo_queue_size', 'type': 'gauge'},
+        {'name': 'ao.redo_rate', 'type': 'gauge'},
+        {'name': 'ao.low_water_mark_for_ghosts', 'type': 'gauge'},
+        {'name': 'ao.filestream_send_rate', 'type': 'gauge'},
+        {'name': 'ao.secondary_lag_seconds', 'type': 'gauge'},
+        # FC - sys.dm_hadr_cluster
+        {'name': 'failover_cluster', 'type': 'tag'},
+    ],
+}
+
+QUERY_FAILOVER_CLUSTER_INSTANCE = {
+    'name': 'sys.dm_os_cluster_nodes',
+    'query': """
+        SELECT
+            NodeName AS node_name,
+            status,
+            status_description,
+            is_current_owner,
+            FC.cluster_name
+        FROM sys.dm_os_cluster_nodes
+        CROSS APPLY (SELECT cluster_name FROM sys.dm_hadr_cluster) AS FC
```

Comment:
I think this is not the join operation you want. This will run `SELECT cluster_name FROM sys.dm_hadr_cluster` for every row in the left table. Here are some candidates for getting the cluster_name on every row:


```sql
        SELECT
            NodeName AS node_name,
            status,
            status_description,
            is_current_owner,
            FC.cluster_name
        FROM sys.dm_os_cluster_nodes
        -- notice the cross *join* which should multiply all rows in the left table against all rows in the right table (top 1)
        CROSS JOIN (SELECT TOP 1 cluster_name FROM sys.dm_hadr_cluster) AS FC
```

```sql
        SELECT
            NodeName AS node_name,
            status,
            status_description,
            is_current_owner,
            FC.cluster_name
        FROM sys.dm_os_cluster_nodes
        JOIN sys.dm_hadr_cluster FC
        WHERE 1=1  -- unconditional left join
```

It will take some experimenting to validate which approach works. They are a little hacky so _please document them well_.